handler/src/main/java/io/netty/handler/ssl/util/SelfSignedCertificate.java

```diff
@@ -332,6 +332,8 @@ public void delete() {
             wrappedBuf.release();
         }
 
+        fqdn = fqdn.replaceFirst("\\*", "X"); // Change asterisk to 'X'
```

Comment:
I think to be safe you want to use something like :

replaceAll("[^\\w.-]", "_");

Reply:
With: `fqdn = fqdn.replaceFirst("\\*", "X");`, Output: `X.shieldblaze.com`
With: `fqdn = fqdn.replaceAll("[^\\w.-]", "_");`, Output: `_.shieldblaze.com`

Both are acceptable in terms of safety. However, we're already appending `_` (underscore) after `keyutil`. We'll end up with 2 underscores.
https://github.com/netty/netty/blob/86765c840cf85284780a0be90191fc530c73ce05/handler/src/main/java/io/netty/handler/ssl/util/SelfSignedCertificate.java#L368

Which one is good? @normanmaurer @chrisvest

Reply:
I think we should use `fqdn.replaceAll("[^\\w.-]", "x")`

Reply:
Done, PTAL.